### PR TITLE
Add a link to the gitea repo in the footer

### PR DESCRIPTION
**Pull request by silverwind**
This adds a link around the git hash in the application version to commit on GitHub. It should be useful for development to know the exact commit one is currently on.

image

Maybe the repo base URL should be extracted to a constant somewhere, but I'm not sure where to put it (maybe `main.go`?).

### CODE REVIEW

**Comment by jolheiser** [comment]
Perhaps the whole version `1.16.0...` could be the link? It looks out of place when the hash is the only linked part imo

**Comment by silverwind** [comment]
@zeripath done.

**Comment by axifive** [comment]
We definitely should get the commit hash from variable without hacks

**Comment by silverwind** [comment]
> We definitely should get the commit hash from variable without hacks

There's no such variable, it's pulled from git on build:

https://github.com/go-gitea/gitea/blob/efdbba44538939d6355cceef59fc18c01bb07630/Makefile#L88

I guess we could leverage our existing git code to do it, but I'm not sure it's really worth the effort. It'd certainly be slower that way.

**Comment by zeripath** [comment]
This won't work on tagged release versions.

**Comment by silverwind** [comment]
I guess tags could be made to link to the tags, page, e.g. https://github.com/go-gitea/gitea/releases/tag/v1.15.2.

**Review by lunny — APPROVED** [review]

**Comment by axifive** [comment]
I added commit hash value from `DRONE_COMMIT` or `git rev-parse HEAD` https://github.com/axifive/gitea/commit/5cc023bd1ebd1cefa7e29db2492b5cf6ac78f094

**Comment by silverwind** [comment]
Thanks, will try to incorporate that later.

File: modules/templates/helper.go [hunk 1]
```diff
@@ -66,6 +67,20 @@ func NewFuncMap() []template.FuncMap {
 		"AppVer": func() string {
 			return setting.AppVer
 		},
+		// render the application version with a link around it
+		// if the version string contains a git commit hash
+		"AppVerLink": func() string {
```
Comment (wxiaoguang) [inline modules/templates/helper.go:72]: The return value of `AppVerLink` can be saved into a variable and cached. It never changes during the lifecycle of the gitea instance, so do not need to call `FindStringSubmatch` and other functions on every page.

Comment (silverwind) [reply modules/templates/helper.go:72]: Yes indeed, I plan to just calculate the link once on app startup and store it in memory.

**Pull request closed without merging**